pkg/k8s/policyhelper/policy.go

```diff
@@ -2,27 +2,64 @@ package policyhelper
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"strings"
 
 	"golang.org/x/exp/slices"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
 	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
 )
 
+var (
+	ErrGroupKind         = errors.New("group/kind error")
+	ErrTargetRefNotFound = errors.New("targetRef not found")
+	ErrTargetRefConflict = errors.New("targetRef has conflict")
+)
+
+type (
+	TargetRef       = gwv1alpha2.PolicyTargetReference
+	ConditionType   = gwv1alpha2.PolicyConditionType
+	ConditionReason = gwv1alpha2.PolicyConditionReason
+)
+
+const (
+	// GEP
+
+	ConditionTypeAccepted = gwv1alpha2.PolicyConditionAccepted
+
+	ReasonAccepted       = gwv1alpha2.PolicyReasonAccepted
+	ReasonInvalid        = gwv1alpha2.PolicyReasonInvalid
+	ReasonTargetNotFound = gwv1alpha2.PolicyReasonTargetNotFound
+	ReasonConflicted     = gwv1alpha2.PolicyReasonConflicted
+
+	// Non-GEP
+
+	ReasonUnknown = ConditionReason("Unknown")
```

Comment:
When should we use this?

Reply:
When reconciler cannot reconcile for other reasons than validation. Bugs, errors in calling k8s api